cli/2024-04-azure-workload-identity.md

```diff
@@ -0,0 +1,264 @@
+
+1. **Azure AD Workload Identity mutating admission webhook (AzWI mutating admission webhook)** is a pod on the Kubernetes cluster that intercepts the pod creation request and injects the signed service account token into the pod. This signed service account token is used by the Azure provider to authenticate with Azure resources.
+
+
+## Objectives
+
+> **Issue Reference: https://github.com/radius-project/radius/issues/7308**
+
+### Goals
+
+* Radius users can configure Azure provider to use Azure workload identity for authentication.
+* Azure workload identity can be configured via interactive experience.
+* Azure workload identity can be configured manually (scriptable).
+* Radius users can deploy and manage Azure resources without using a Entra ID Application client secret.
+
+### Non-goals
+
+* **Azure Managed Identity support**
+    - Managed Identity support for AKS is a much older feature, and Workload Identity is the replacement. We're not planning to support Managed Identity, as any user should be able to upgrade to Workload Identity.
+
+* **AWS Workload Identity (IRSA) support**
+    - Will be tracked as separate work: https://github.com/radius-project/radius/issues/7618
+
+### User scenarios
+
+#### User story 1: As an infrastructure operator, I can configure Radius to use Azure workload identity via interactive experience.
+
+1. Install the `rad` CLI.
+
+1. run `rad init --full` to begin the interactive experience
+    ```
+    rad init --full
+    ```
+
+1. Enter the environment name, namespace and select "yes" to add a cloud provider and select "Azure" to add an Azure cloud provider.
+    ```
+    Select your cloud provider:
+    > 1. Azure
+      2. AWS
+      3. [back]
+    ```
+
+1. Enter or use the Azure subscription and resource group name. Select  "Workload Identity" for the Azure cloud provider.
+    ```
+    Select the identity for the Azure cloud provider 
+    1. Service principal                               
+    > 2. Workload Identity    
+    ```
+
+1. Enter the Entra ID Application Client ID and Tenant ID.
+    ```
+    Please follow the guidance at aka.ms/rad-workload-identity to set up workload identity for Radius.
+
+    Enter the Client ID of the Entra ID Application that you have created for Radius.
+
+    > Enter the Client ID:
+
+    Enter the Tenant ID of the Entra ID Application that you have created for Radius.
+
+    > Enter the Tenant ID:
+    ```
+
+Radius is now configured to use Azure workload identity for authentication.
+
+#### User story 1: As an infrastructure operator, I can configure Radius to use Azure workload identity without an interactive experience.
+
+1. Install the `rad` CLI.
+
+1. Install Radius using `rad install kubernetes`
+    ```
+    rad install kubernetes --set global.azureWorkloadIdentity.enabled=true
```

Comment:
We should consider adding a `rad install kubernetes` landing page which includes a "what do I do next"?

/cc @willtsai